src/Webapi/Temporal/Webapi__Temporal__ZonedDateTime.res

```diff
@@ -0,0 +1,152 @@
+type t = Webapi__Temporal__Types.zonedDateTime
+
+// @new external make: (~epochNanoseconds: BigInt.t, ~timeZone: TimeZone.t) => t = "Temporal.ZonedDateTime"
+// @new external makeWithCalendar: (~epochNanoseconds: BigInt.t, ~timeZone: TimeZone.t, ~calendar: Calendar.t) => t = "Temporal.ZonedDateTime"
+
+module ZonedDateTimeInit = {
+  type t
+
+  @obj
+  external make: (
+    ~timeZone: Webapi__Temporal__Types.timeZone=?,
+    ~year: int=?,
+    ~month: int=?,
+    ~day: int=?,
+    ~minute: int=?,
+    ~second: int=?,
+    ~millisecond: int=?,
+    ~microsecond: int=?,
+    ~nanosecond: int=?,
+  ) => t = ""
+}
+
+module FromOptions = {
+  type t
+
+  @obj
+  external make: (
+    ~overflow: [#constrain | #reject]=?,
+    ~disambiguation: [#compatible | #earlier | #later | #reject]=?,
+    ~offset: [#use | #ignore | #prefer | #reject]=?,
+  ) => t = ""
+}
+
+@scope("Temporal.ZonedDateTime") @val external fromString: string => t = "from"
+@scope("Temporal.ZonedDateTime") @val
+external fromStringWithOptions: (string, FromOptions.t) => t = "from"
```

Comment:
I've taken different approaches for option bags depending on the optionality of the properties. My logic goes as follows:

- If there is only one property, use an object type. Mark it as required even if the API accepts an empty object
- If there is more than one property but they're all required, use an object type
- If there is more than one property and one or more are optional, use an option bag module

Is this consistent with other bindings? If so, would be good to document this in some style guide.

Reply:
Record types are best when all properties are required, imo. They have the advantage of looking like objects when calling the function; the `@obj` system was created a long time ago and is now only used as a bit of a hack to omit unspecified properties instead of setting them to `undefined`.

And yes now that we're getting more contributors a style guide is probably worth starting. It might make a nice binding cheat-sheet too.